lib/network.py

```diff
@@ -1933,6 +1939,26 @@ def get_proxies(self):
             return proxies
         return None
 
+    def on_bad_certificate(self, server, certificate):
+        if server in self.bad_certificate_servers:
+            return
+        self.bad_certificate_servers[server] = certificate
+        self.server_list_updated()
+
+    def remove_bad_certificate(self, server):
+        if server not in self.bad_certificate_servers:
+            return
+        del self.bad_certificate_servers[server]
+        self.server_list_updated()
+
+    def remove_pinned_certificate(self, server):
+        cert_file = self.bad_certificate_servers.get(server)
+        if cert_file:
+            os.unlink(cert_file)
```

Comment:
I'm always paranoid about weird setups -- this is unlikely but do you think it's worth `try`, catch `OSError` here around the `os.unlink` and if that happens, returning early?

Users have extremely weird filesystem setups -- sometimes using DropBox.. and other bizarro stuff.. which is why I have fund it's just better to be safe than sorry esp. with "explodey" python.

Reply:
Updated. I also added the same checks to the other places we remove certificates.